File: 30-Days-of-Code/Day-4-Class-vs.-Instance/Lua/solution.lua
local Person = { };
Person.__index = Person
 
function Person.new (initialAge)
   local self = setmetatable({ },Person);
   self.age = initialAge
    
   if initialAge < 0 then
        self.age = 0
        print("Age is not valid, setting age to 0.")
   end
        
   return self;
end
function Person:amIOld()
    if self.age < 13 then
        print("You are young.") 
    elseif self.age >= 13 and self.age < 18 then
        print("You are a teenager.")
    else
        print("You are old.")
    end 
end
function Person:yearPasses ()
   self.age = self.age + 1
end

local read = io.read;
local T = read'*n';
for i = 1,T do
   local age = read'*n';
   local p = Person.new(age);
   p:amIOld();
   for j = 1,3 do
      p:yearPasses();
   end
   p:amIOld();
   print"";
end

File: 30-Days-of-Code/Day-4-Class-vs.-Instance/PHP/solution.php
<?php
class Person{
    public $age;
   public function __construct($initialAge){
          $this->age = $initialAge;
              
          if ($initialAge < 0) {
              $this->age = 0;
              echo "Age is not valid, setting age to 0.\n";
          }

    }
   public  function amIOld(){
        if ($this->age < 13) {
            echo "You are young.\n";
        } elseif ($this->age >= 13 and $this->age < 18) {
            echo "You are a teenager.\n";
        } else {
            echo "You are old.\n";
        }

    }
   public  function yearPasses(){
          $this->age++;
   }
   
      
}

$T = intval(fgets(STDIN));
 for($i=0;$i<$T;$i++){
     $age=intval(fgets(STDIN));
     $p=new Person($age);
     $p->amIOld();
     for($j=0;$j<3;$j++){
         $p->yearPasses();
     }
     $p->amIOld();
     echo "\n";
         
 }
?>
